Empire/Empire.cpp

```diff
@@ -166,6 +167,11 @@ namespace {
                 elem.allocated_rp = 0.0f;
             }
         }
+
+        if (total_RPs_spent >= RPs - EPSILON)
+            return; // don't further allocate if nothing left to spend
```

Comment:
The place these lines are located, after the end of the loop, it seems they are only avoiding the logging line that reports how much RP was spent, not actually stopping any allocation the way the comment indicates is intended.

If I move these lines up into the bottom of the loop so that they stop further allocation, then they cause a UI error:   start a new game, queue up a cheap tech like Planet Eco and then an expensive one like Radar.  Now shift Radar to first spot so that it soaks up all the available RP.  Planet Eco still shows as if it is getting RP allocated to it since the loop exited as soon as Radar used up the RP.  (The server prevents any actual overallocation during turn processing though.)